### PR TITLE
feature: add logging across APIs

### PR DESCRIPTION
**Pull request by sasha-gitg**

### CODE REVIEW

File: google/cloud/aiplatform/lro.py [hunk 1]
```diff
@@ -1,109 +0,0 @@
...
-
-import proto
-
-from google.api_core import operation as ga_operation
-from google.cloud.aiplatform import base
-from typing import Optional, Callable
-
-
-class LRO:
-    """A handler for operation futures."""
```
Comment (vinnysenthil) [inline google/cloud/aiplatform/lro.py:26]: Is the class being removed to lack of use?

File: google/cloud/aiplatform/base.py [hunk 2]
```diff
@@ -19,16 +19,137 @@
...
+        Args:
+            noun (str): Noun the action acts on against the resource. Can be empty string.
+            action (str): Action started against resource. ie: "Deployed". Can be empty string.
+            cls (AiPlatformResourceNoun):
+                Resource noun object the action is acting against.
+            resource_noun_obj (AiPlatformResourceNoun):
+                Resource noun object the action is acting against
+        """
+        self._logger.info(
+            f"{self.__class__.__name__} {noun} {action}. Resource name: {resource_noun_obj.resource_name}"
```
[the remark this answers is not in the document]
Comment (vinnysenthil) [reply google/cloud/aiplatform/base.py]: Never mind, looks like you fixed this in a follow up commit :)

**Review by vinnysenthil — APPROVED** [review]
LGTM, just a few questions and one docstring fix

File: google/cloud/aiplatform/datasets/dataset.py [hunk 3]
```diff
@@ -309,11 +315,29 @@ def _create_and_import(
...
+        # _LOGGER.info(f"Import {self.__class__.__name__} data backing LRO: {import_lro.operation.name}")
+
+        _LOGGER.log_action_started_against_resource_with_lro(
+            "Import", "data", self.__class__, import_lro
+        )
+
+        import_lro.result()
+
+        # _LOGGER.info(f"{self.__class__.__name__} data imported. Resource name: {self.name}")
+        _LOGGER.log_action_completed_against_resource("data", "imported", self)
```
[the remark this answers is not in the document]
Comment (sasha-gitg) [reply google/cloud/aiplatform/datasets/dataset.py:339]: Good catch. Removing.

File: google/cloud/aiplatform/lro.py (hunk 1 above)
Comment (sasha-gitg) [reply google/cloud/aiplatform/lro.py:26]: Yes, it's currently unused throughout the SDK. We can restore it if needed.

File: google/cloud/aiplatform/base.py [hunk 4]
```diff
@@ -19,16 +19,137 @@
...
+    def log_action_completed_against_resource(
+        self, noun: str, action: str, resource_noun_obj: "AiPlatformResourceNoun"
+    ):
+        """Logs action completed against resource.
+
+        Args:
+            noun (str): Noun the action acts on against the resource. Can be empty string.
+            action (str): Action started against resource. ie: "Deployed". Can be empty string.
+            cls (AiPlatformResourceNoun):
+                Resource noun object the action is acting against.
```
[the remark this answers is not in the document]
Comment (sasha-gitg) [reply google/cloud/aiplatform/base.py:112]: Removed. Thanks!

**Pull request merged**